internal/pkg/endpoint/endpoint.go

```diff
@@ -15,50 +15,77 @@
 package endpoint
 
 import (
+	"context"
 	"fmt"
 	"os"
+	"sync"
 	"time"
 
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/interfaces"
 	"github.com/edgexfoundry/go-mod-registry/registry"
 )
 
 type Endpoint struct {
+	ctx            context.Context
+	wg             *sync.WaitGroup
 	RegistryClient *registry.Client
+	serviceKey     string // The key of the service as found in the service registry (e.g. Consul)
+	path           string // The path to the service's endpoint following port number in the URL
+	interval       int    // The interval in milliseconds governing how often the client polls to keep the endpoint current
+
+}
+
+func New(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	registryClient *registry.Client,
+	serviceKey string,
+	path string,
+	interval int) *Endpoint {
+
+	return &Endpoint{
+		ctx:            ctx,
+		wg:             wg,
+		RegistryClient: registryClient,
+		serviceKey:     serviceKey,
+		path:           path,
+		interval:       interval,
+	}
 }
 
-func (e Endpoint) Monitor(params types.EndpointParams) chan string {
-	ch := make(chan string, 1)
-	ch <- e.fetch(params)
+func (e Endpoint) Monitor() chan interfaces.URLStream {
+	ch := make(chan interfaces.URLStream, 1)
+	url, err := e.buildURL()
+	if err != nil {
+		_, _ = fmt.Fprintln(os.Stdout, err.Error())
+	}
+	ch <- interfaces.URLStream(url)
+
+	e.wg.Add(1)
 	go func() {
+		defer e.wg.Done()
 		for {
-			url, err := e.buildURL(params)
+			url, err := e.buildURL()
 			if err != nil {
 				_, _ = fmt.Fprintln(os.Stdout, err.Error())
 			}
-			ch <- url
-			time.Sleep(time.Millisecond * time.Duration(params.Interval))
+			ch <- interfaces.URLStream(url)
+			time.Sleep(time.Millisecond * time.Duration(e.interval))
+
+			// use ctx to drop out of infinite when ctx indicates done().
```

Comment:
You pass ctx in and store it in the receiver but don't actually use it in your goroutine to break out if the context is Done().  Please update your implementation to exit when ctx is cancelled.